### PR TITLE
Take two: close #186 (Automatic) dark mode

### LINKED ISSUE #186: (Automatic) dark mode
## Current experience

image

on GNOME (Fedora/Linux) with Firefox 68 here.

In contrast to Firefox' own/internal popups:
image

### Problems

* A common issue I see on websites (but possibly also an issue of Firefox as it should not handle it like this?), is that input fields inherit the system's white font color by default.
* Generally, the white popup stands out a lot…

## Expected

1. Possibly report the first thing as an issue in Firefox. (don't know how it looks on other OSes)
2. Implement a dark mode that is triggered, when the system uses a dark theme…

To do so, there is this **new fancy** CSS media query: https://developer.mozilla.org/docs/Web/CSS/@media/prefers-color-scheme aka `prefers-color-scheme`
But should the need arise, you can use this JS API: https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia

Note: There **must not** be any manual selection of this dark mode. I'd consider this an unnecessary option…

Of course, as always, adhere to the [Photon design](https://design.firefox.com/photon).

3. (optional) As for the QR code itself, that would obviously usually also need to be inverted. So it could be an idea to also change the default colors of the QR code, if it is in dark mode. (of course, do not overwrite user settings though)
  However, this requires some JS changes (which is possible as explained above), but actually even worse: Many QR code scanners read them. (see https://github.com/rugk/offline-qr-code/issues/20)
  So, if we do so, we definitively need to implement https://github.com/rugk/offline-qr-code/issues/20 first. Or find some way that is still scannable, but looks decent in the dark mode.

### Good first issue

This is a good first issue, as it is likely a CSS only change. (And hopefully with not-so many CSS changes.)

Also any more ideas on this are appreciated. :smiley:

### PR DESCRIPTION
**Pull request by davidfloyd91**
Fixes https://github.com/rugk/offline-qr-code/issues/186

Incorporates suggested changes from https://github.com/rugk/offline-qr-code/pull/201

### CODE REVIEW

File: src/common/modules/data/DefaultSettings.js [hunk 1]
```diff
@@ -4,6 +4,9 @@
  * @module data/DefaultSettings
  */
 
+// checks for OS dark theme
+const darkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
```
Comment (rugk) [inline src/common/modules/data/DefaultSettings.js]:
```suggestion
const isDarkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
```

File: src/common/modules/data/DefaultSettings.js [hunk 2]
```diff
@@ -16,7 +19,7 @@ const defaultSettings = Object.freeze({
     popupIconColored: false,
     qrCodeType: "svg",
     qrColor: "#0c0c0d",
-    qrBackgroundColor: "#ffffff",
+    qrBackgroundColor: (darkTheme ? "#d7d7db" : "#ffffff"),
```
Comment (rugk) [inline src/common/modules/data/DefaultSettings.js]: GitHub makes funny diffs, BTW:
image

Anyway, to get serious, you missed to document that "magic value"=color there now…

Comment (rugk) [reply src/common/modules/data/DefaultSettings.js]:
```suggestion
    qrBackgroundColor: (darkTheme ? "#d7d7db" : "#ffffff"), // dark uses Firefox Photon's grey-30
```

Comment (rugk) [reply src/common/modules/data/DefaultSettings.js]:
```suggestion
    qrBackgroundColor: (isDarkTheme ? "#d7d7db" : "#ffffff"),
```

also obviously

**Comment by rugk** [comment]
Also tested it, here is a sceenshot:
image

So… yeah… looks pretty decent, I guess.

File: src/common/modules/data/DefaultSettings.js (hunk 1 above)
Comment (davidfloyd91) [reply src/common/modules/data/DefaultSettings.js]: just push this up. couldn't do it here in Github for some reason

File: src/common/modules/data/DefaultSettings.js (hunk 2 above)
Comment (davidfloyd91) [reply src/common/modules/data/DefaultSettings.js]: pushed this up too

Comment (rugk) [reply src/common/modules/data/DefaultSettings.js]: Thanks yeah, just clicking "resolve conversation" should do it. :smile:

**Pull request merged**